include/programs.h

```diff
@@ -36,16 +36,26 @@ class CommandLine {
 
 	const char *GetFileName() { return file_name.c_str(); }
 
-	bool FindExist(char const * const name,bool remove=false);
-	bool FindInt(char const * const name,int & value,bool remove=false);
-	bool FindString(char const * const name,std::string & value,bool remove=false);
-	bool FindCommand(unsigned int which,std::string & value);
-	bool FindStringBegin(char const * const begin,std::string & value, bool remove=false);
-	bool FindStringRemain(char const * const name,std::string & value);
+	bool FindExist(char const *const name, bool remove = false);
+	bool FindSlashOption(const std::string &name, bool remove = false);
+	bool FindDashOption(const std::string &name, bool remove = false);
+	bool FindOption(const std::string &name, bool remove = false);
+	bool FindInt(char const *const name, int &value, bool remove = false);
+	bool FindIntOption(const std::string &name, int &value, bool remove = false);
+	bool FindString(char const *const name, std::string &value, bool remove = false);
+	bool FindStringOption(const std::string &name,
+	                      std::string &value,
+	                      bool remove = false);
+	bool FindCommand(unsigned int which, std::string &value);
+	bool FindStringBegin(char const *const begin,
+	                     std::string &value,
+	                     bool remove = false);
+	bool FindStringRemain(char const *const name, std::string &value);
 	bool FindStringRemainBegin(char const *const name, std::string &value);
-	bool GetStringRemain(std::string & value);
-	int GetParameterFromList(const char* const params[], std::vector<std::string> & output);
-	void FillVector(std::vector<std::string> & vector);
+	bool GetStringRemain(std::string &value);
+	int GetParameterFromList(const char *const params[],
+	                         std::vector<std::string> &output);
```

Comment:
The names of these functions are nice and self-documenting:
 -  `FindDashOption("version")`: the "Dash" tells it looks for dosbox --version` instead of `dosbox /version`
 -  `FindSlashOption("version")`: the "Slash" tells me it looks for `Z:\> COMMAND /VERSION` instead of `Z:\> COMMAND --VERSION`

The other functions are less obvious what will happen:
 -  `FindOption("h")`: will that scan for `-h` or `/h` options? Also, it takes in a `std::string`, so can I feed it long options too, like `FindOption("help")`?
 - `FindStringOption` and `FindIntOption` tell me get option arguments argument, but what prefix do they handle? Will `FindStringOption("ide", drive, false)` look for `/ide` or `-ide`? Do these handle long and short options too, like `/C ABC`?

There's a lot of good rework here @Wengier, hopefully we can make the names more self documenting. 

To help all of us work with the code, I would also like comment block summarizing how they should be used.

Reply:
👀 
looks fun, like a small headache

its good to see someone start tackling this

i remember our discussion about loguru switches and the new runtime switches when 0.78.0 was released

was runtime switching ported from dosbox-x?  
eg: `dosbox -machine`